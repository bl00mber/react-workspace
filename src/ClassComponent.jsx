class ClassComponent extends React.Component {
  render() {
    return <h3>{this.props.name}</h3>;
  }
}
