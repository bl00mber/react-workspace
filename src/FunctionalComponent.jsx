import React from 'react';
import PropTypes from 'prop-types';

const FunctionalComponent = props => (
  <h3>{props.name}</h3>
);

FunctionalComponent.propTypes = {
  name: PropTypes.string.isRequired,
};

export default FunctionalComponent;
