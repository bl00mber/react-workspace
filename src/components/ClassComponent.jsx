import React from 'react';
import PropTypes from 'prop-types';

class ClassComponent extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      test: true,
    };
  }

  render() {
    return this.state.test && <h3>{this.props.name}</h3>;
  }
}

ClassComponent.propTypes = {
  name: PropTypes.string.isRequired,
};

export default ClassComponent;
