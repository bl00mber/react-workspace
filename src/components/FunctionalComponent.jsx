import React from 'react';
import PropTypes from 'prop-types';

const FunctionalComponent = ({ name }) => (
  <h3>{name}</h3>
);

FunctionalComponent.propTypes = {
  name: PropTypes.string.isRequired,
};

export default FunctionalComponent;
