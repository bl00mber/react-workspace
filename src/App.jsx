import React from 'react';
import { Provider } from 'react-redux';

import FunctionalComponent from '@/components/FunctionalComponent';
import ClassComponent from '@/components/ClassComponent';
import Clock from '@/components/Clock';
import CustomForm from '@/components/CustomForm';
import AddTodoComponent from '@/containers/AddTodo';
import VisibleTodoList from '@/containers/VisibleTodoList';
import Footer from '@/components/Footer';
import Toggle from '@/containers/Toggle';
import Calculator from '@/containers/Calculator';
import PostList from '@/containers/PostList';

import { store } from '@/store';

const App = () => (
  <Provider store={store}>
    <div>
      <h1>Hello, React Workspace!</h1>
      <FunctionalComponent name="Functional Component" />
      <ClassComponent name="Class Component" />
      <Clock />
      <CustomForm />
      <div>
        <AddTodoComponent />
        <VisibleTodoList />
        <Footer />
      </div>
      <Toggle />
      <Calculator />
      <PostList />
    </div>
  </Provider>
);


// The only way to mutate the internal state is to dispatch an action.
// The actions can be serialized, logged or stored and later replayed.
store.dispatch({ type: 'INCREMENT' });
// 1
store.dispatch({ type: 'INCREMENT' });
// 2
store.dispatch({ type: 'DECREMENT' });
// 1


export default App;
