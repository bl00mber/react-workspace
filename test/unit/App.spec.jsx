import React from 'react';
import { Route } from 'react-router-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import App from '@/App';
import FunctionalComponent from '@/components/FunctionalComponent';
import ClassComponent from '@/components/ClassComponent';

describe('App', () => {
  test('should match enzyme shallow snapshot', () => {
    const wrapper = shallow(<App />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('should render routed components', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.find(Route).at(0).props().render())
      .toEqual(<FunctionalComponent name="Functional Component" />);
    expect(wrapper.find(Route).at(1).props().render())
      .toEqual(<ClassComponent name="Class Component" />);
  });
});
