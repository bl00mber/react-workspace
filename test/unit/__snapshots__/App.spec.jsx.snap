// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match enzyme shallow snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <div>
    <h1>
      Hello, React Workspace!
    </h1>
    <FunctionalComponent
      name="Functional Component"
    />
    <ClassComponent
      name="Class Component"
    />
    <Clock />
    <CustomForm />
    <div>
      <Connect(AddTodo) />
      <Connect(TodoList) />
      <Footer />
    </div>
    <Connect(Toggle) />
    <Connect(Calculator) />
    <Connect(PostList) />
  </div>
</Provider>
`;
